### PR TITLE
LGA: Sign In Page and Modal Alert Box Addition

### PR DESCRIPTION
**Pull request by moanu404**
## Summary

- Adding an alert box to the sign in page to signify the sunset of ds logon and mhv
- This will be rolled out to both the sign in modal and the sign in page
- This solution has been discussed by the designers according to this [figma](https://www.figma.com/file/8vobVoSl1UqLnXsni3rcvO/Login.gov-Adoption%3A-Design-Flows?type=design&node-id=2601-33&mode=design&t=D0zVP0WKPdJBpCTc-0) 
- Part of the LGA Adoption team 
- Once 100 percent rollout is complete this flipper will be removed - tentatively end of September

## Related issue(s)

- _Link to ticket created in va.gov-team repo_
department-of-veterans-affairs/va.gov-team#62298

## Testing done

-  Sign In Page was visible without alert modal
-  Flipper turned on
- Alert is now visible 
- Flipper turned off
-  Alert not visible anymore
-  Added unit tests

### Sign in Page 
Desktop: 
image

Mobile:
image

### Sign In Modal

Desktop flow-

https://github.com/department-of-veterans-affairs/vets-website/assets/137448668/e9a36a73-6325-4f84-914d-05b8643e79fa

Mobile screenshots- 
image

image

## What areas of the site does it impact?
Sign In Page and Modal 

### Quality Assurance & Testing

- [x] I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [x] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [x] Linting warnings have been addressed
- [ ] Documentation has been updated ([link to documentation](#) \*if necessary)
- [x] Screenshot of the developed feature is added
- [ ] [Accessibility testing](https://depo-platform-documentation.scrollhelp.site/developer-docs/wcag-2-1-success-criteria-and-foundational-testing) has been performed

### Error Handling

- [x] Browser console contains no warnings or errors.
- [ ] Events are being sent to the appropriate logging solution
- [ ] Feature/bug has a monitor built into Datadog or Grafana (if applicable)

### Authentication

- [ ] Did you login to a local build and verify all authenticated routes work as expected with a test user

### :warning: Team Sites (only applies to modifications made to the VA.gov header) :warning:

- [ ] The vets-website header does not contain any web-components
- [ ] I used the [proxy-rewrite steps](https://github.com/department-of-veterans-affairs/vets-website/tree/main/src/applications/proxy-rewrite#local-dev) to test the injected header scenario
- [ ] I reached out in the `#sitewide-public-websites` Slack channel for questions

### CODE REVIEW

**Review by powellkerry — DISMISSED** [review]

**Review by joeniquette — CHANGES_REQUESTED** [review]
Before this is pushed out, we need to get approval from the business owners of our USiP clients unless you change this modal change to only apply to va.gov USiP. Happy to discuss the steps of either process. Changing text on our USiP here will impact OCC mobile apps, MHV, Cerner, and VA flagship mobile app.

**Comment by moanu404** [comment]
> Before this is pushed out, we need to get approval from the business owners of our USiP clients unless you change this modal change to only apply to va.gov USiP. Happy to discuss the steps of either process. Changing text on our USiP here will impact OCC mobile apps, MHV, Cerner, and VA flagship mobile app.

Sure @joeniquette let me check with the rest of my team if we have approvals already or not. Is there any documentation you may be able to point me to or we can sync whenever is convenient for you?

**Comment by moanu404** [comment]
> Before this is pushed out, we need to get approval from the business owners of our USiP clients unless you change this modal change to only apply to va.gov USiP. Happy to discuss the steps of either process. Changing text on our USiP here will impact OCC mobile apps, MHV, Cerner, and VA flagship mobile app.

Just chatted and these changes will only be applicable to VA.gov at the moment while we work through approvals with Unified comms and socialize upcoming changes with owners on the business side. And this is to be tested on staging for a bit until rollout anyways. Would love to setup some time and share knowledge about rolling this out to other entities and such :)

**Comment by joeniquette** [comment]
My calendar is in my slack profile, grab anything thats open and works for you. I'll also want to make sure Alex can attend.

File: src/platform/user/authentication/components/LoginActions.jsx
Comment (asg5704) [inline src/platform/user/authentication/components/LoginActions.jsx:18]: Opt for `const showSignInPageAndModalExperiment = useSelector(....)` inside

File: src/platform/user/authentication/components/LoginActions.jsx
Comment (asg5704) [inline src/platform/user/authentication/components/LoginActions.jsx:88]: Opt for `useSelector` instead of `mapStateToProps`

File: src/platform/user/authentication/components/LoginActions.jsx
Comment (asg5704) [inline src/platform/user/authentication/components/LoginActions.jsx:45]: `externalApplication` is the prop you want to check for all the various USiP applications.

1. If `externalApplication` is null, we treat it as the default (or normal VA.gov login)
2. The list of `externalApplications` can be found in the `user/authentication/constants.js` file

File: src/platform/user/tests/authentication/components/loginActions.unit.spec.js
Comment (asg5704) [inline src/platform/user/tests/authentication/components/loginActions.unit.spec.js:16]: 1. Create tests so that you can toggle between the 2 states and confirm your modal shows appropriately.
2. Create another test to loop through all of the external applications of USiP and confirm that they do not show the modal.

**Review by asg5704 — CHANGES_REQUESTED** [review]
See comments above.

File: src/platform/user/authentication/components/LoginActions.jsx
Comment (moanu404) [reply src/platform/user/authentication/components/LoginActions.jsx:88]: Hey @asg5704 thank you for the feedback. I am pretty new and not that versed with Redux. I did use `mapStateToProps` because of its usage in other parts of the codebase which used it so I followed the pattern of a few other components. I am happy to learn and use - `useSelector` but wondering if it is to be established as a new norm/pattern in the future or the merits of it?

Comment (moanu404) [reply src/platform/user/authentication/components/LoginActions.jsx:88]: `script/github-actions/daily-product-scan/tests/mocks/applications/app-1/HealthCareApp.jsx` for instance here. Also are you talking about just moving the feature flag bit to its own `selectors.js` file?

Comment (asg5704) [reply src/platform/user/authentication/components/LoginActions.jsx:88]: Hey @moanu404

So just some background a lot of the codebase uses old React patterns that while still work could become deprecated in the future as React works towards more hook-based patterns. So the main reason why I recommend using `useSelector` instead of `mapStateToProps` is because `mapStateToProps + connect` passes down multiple values as props with the higher-order component where as `useSelector` looks for one value and does not bind it to props so you get a tiny performance bump.

As for the selector, you can put it in the `platform/user/authentication/selectors.js` file to keep the component relatively clean since there are already the `toggleValues` imports.

If you have any questions, etc. just reach back out.

File: src/platform/user/authentication/components/LoginActions.jsx
Comment (asg5704) [inline src/platform/user/authentication/components/LoginActions.jsx:43]: Change `==` to `===` (JS best practices. == will try to cast to a type if possible, whereas === will do strict type checking)

Comment (moanu404) [reply src/platform/user/authentication/components/LoginActions.jsx:43]: Interesting, @asg5704 when I `console.log(externalApplication === null)` it returns false locally. Is this intended behavior, I think on `http://localhost:3001/?next=loginModal&oauth=true` externalApplication is undefined? 

Not sure if there is a difference between local vs other env's..

Comment (asg5704) [reply src/platform/user/authentication/components/LoginActions.jsx:43]: Ah sorry @moanu404 I was looking strictly at the code. What you can do instead is `!externalApplication && rest of yoru code` this will check for `null` or `undefined` my apologies.

Comment (moanu404) [reply src/platform/user/authentication/components/LoginActions.jsx]: Copy that! No worries haha, I was totally on autopilot from ruby with only using `==` . I just made the change, TIA for looking :)

**Review by asg5704 — DISMISSED** [review]

**Pull request closed without merging**
